lib/util.js

```diff
@@ -126,3 +128,24 @@ exports.getParserServices = context => {
     }
     return context.parserServices;
 };
+
+/**
+ * @param {ts.SyntaxKind} syntaxKind A TS syntax kind
+ * @returns {string} The corresponding string representation
+ */
+function getESTreeType(syntaxKind) {
+    const tsSyntaxKindName = ts.SyntaxKind[syntaxKind];
+
+    return parser.Syntax[tsSyntaxKindName] || `TS${tsSyntaxKindName}`;
```

Comment:
this is not always a case, sometimes names do change, i don't think it's good idea..

i think its better to import types from 

https://github.com/JamesHenry/typescript-estree/blob/master/src/ast-node-types.ts

Reply:
That is what I'm doing; `parser.Syntax` is pulling from there, it just seemed right to do it through `typescript-eslint-parser` as opposed to directly referencing `typescript-estree`. The second case is just backup for nodes missing from the file you linked.

Reply:
there are also nodes not present in ts, like `TSTypeParameterDeclaration`, `TSTypeParameterInstantiation`

Reply:
Oh hmm, I didn't see your edit. Let me see if that works better.

Reply:
also in some rare cases this using conversion `ts.SyntaxKind[syntaxKind]` will return you invalid name 

this enum contains also `FirstAssignment`, `LastAssignment` and so on, it will convert it to last string from enum with number value

instead of `TemplateTail` you will get `LastTemplateToken`

Reply:
`TemplateTail` is not present in estree its called `TemplateElement` :)

Reply:
Yup, I totally see your point. I'll put together a different approach later today.